Twig/NormalizingNodeVisitor.php

```diff
@@ -26,14 +26,14 @@
  *
  * @author Johannes M. Schmitt <schmittjoh@gmail.com>
  */
-class NormalizingNodeVisitor implements \Twig_NodeVisitorInterface
+class NormalizingNodeVisitor extends \Twig_BaseNodeVisitor
 {
     /**
      * @param \Twig_NodeInterface $node
      * @param \Twig_Environment $env
      * @return \Twig_NodeInterface
      */
-    public function enterNode(\Twig_NodeInterface $node, \Twig_Environment $env)
```

Comment:
Yes, I know. 

I did begin by trying what you have suggested (created new visitors for Twig2, that have the right signature for methods). Turns out that the twig 1 classes are blocking this, as the classes will still be parsed by PHP, and the implementation does not match the (twig2) interface. When Twig2 is installed, only the Twig2 specific classes will be valid, and the Twig1 ones will be invalid. And the other way around with Twig1 installed.

I have tried to think about ways around this, but couldn't figure out any. Then I stumbled upon how this had been covered in Twig itself, with this abstract class that changes `enterNode` and `leaveNode` to final methods, and adds the abstract classes.

Or am I missing something obvious ?